scripts/generate_aviti_run_manifest.py

```diff
@@ -0,0 +1,323 @@
+                    ["Index1", "Index1_rc"],
+                ):
+                    for b2, _b2 in zip(
+                        [row_comp["Index2"], revcomp(row_comp["Index2"])],
+                        ["Index2", "Index2_rc"],
+                    ):
+                        flips.append(
+                            (
+                                distance(a1, b1) + distance(a2, b2),
+                                f"{a1}-{a2} {b1}-{b2}",
+                                f"{_a1}-{_a2} {_b1}-{_b2}",
+                            )
+                        )
+        dist, compared_seqs, flip_conf = min(flips, key=lambda x: x[0])
+
+    else:
+        dist = distance(
+            row["Index1"] + row["Index2"], row_comp["Index1"] + row_comp["Index2"]
+        )
+        compared_seqs = (
+            f"{row['Index1']}-{row['Index2']} {row_comp['Index1']}-{row_comp['Index2']}"
+        )
+
+    if dist <= dist_warning_threshold:
+        # Build a warning message for the pair
+        warning_lines = [
+            f"Hamming distance {dist} between {row['SampleName']} and {row_comp['SampleName']}"
+        ]
+        # If the distance is derived from a flip, show the original and the flipped conformation
+        if check_flips:
+            warning_lines.append(
+                f"Given: {row['Index1']}-{row['Index2']} <-> {row_comp['Index1']}-{row_comp['Index2']}"
+            )
+            warning_lines.append(f"Distance: {dist} when flipped to {flip_conf}")
+        # If the index lengths are equal, add a simple visual representation
+        if len(row["Index1"]) + len(row["Index2"]) == len(row_comp["Index1"]) + len(
+            row_comp["Index2"]
+        ):
+            warning_lines.append(show_match(*compared_seqs.split()))
+
+        warning = "\n".join(warning_lines)
+        logging.warning(warning)
+
+        # For identical collisions, kill the process
+        if dist == 0:
+            raise AssertionError("Identical indices detected.")
+
+
+def revcomp(seq: str) -> str:
+    """Reverse-complement a DNA string."""
+    return seq.translate(str.maketrans("ACGT", "TGCA"))[::-1]
+
+
+def show_match(seq1: str, seq2: str) -> str:
+    """Visualize base-by-base match between sequences of equal length."""
+
+    assert len(seq1) == len(seq2)
+
+    m = ""
+    for seq1_base, seq2_base in zip(seq1, seq2):
+        if seq1_base == seq2_base:
+            m += "|"
+        else:
+            m += "X"
+
+    lines = "\n".join([seq1, m, seq2])
+    return lines
+
+
+def safe_string(s: str) -> str:
+    """Wrap a string in quotes if it contains commas."""
+    if "," in s:
+        return f'"{s}"'
+    else:
+        return s
+
+
+@epp_decorator(script_path=__file__, timestamp=TIMESTAMP)
+def main(args: Namespace):
+    lims = Lims(BASEURI, USERNAME, PASSWORD)
+    process = Process(lims, id=args.pid)
+
+    file_name = f"AVITI_run_manifest_{process.id}_{TIMESTAMP}_{process.technician.name.replace(' ','')}.csv"
```

Comment:
This may need to be fixed later. We will prefer to have the unique flowcell ID here.

Reply:
[0cbddfc](https://github.com/SciLifeLab/scilifelab_epps/pull/342/commits/0cbddfc60c54a6f85b6e5b2e6483a12c1394f275)